src/index.js

```diff
@@ -1,21 +1,24 @@
 // eslint-disable-next-line no-unused-vars
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useMemo } from 'react';
 import { Dimensions } from 'react-native';
 
+export const ORIENTATION = {
+  LANDSCAPE: 'landscape',
+  PORTRAIT: 'portrait',
+};
+
+function getWindowOrientation() {
+  const { width, height } = Dimensions.get('window');
+  return height >= width ? ORIENTATION.PORTRAIT : ORIENTATION.LANDSCAPE;
+}
 function useDeviceOrientation() {
-  const [deviceOrientation, setDeviceOrientation] = useState(null);
+  const initialValue = useMemo(() => getWindowOrientation(), []);
+  const [deviceOrientation, setDeviceOrientation] = useState(initialValue);
```

Comment:
I think you can directly pass a function to `useState`. This way it would only be called once and `useMemo` won't be required. [Docs](https://reactjs.org/docs/hooks-reference.html#lazy-initial-state)

```suggestion
  const [deviceOrientation, setDeviceOrientation] = useState(getWindowOrientation);
```

Can you test this once?